torch/csrc/jit/tensorexpr/ir_printer.cpp

```diff
@@ -22,13 +22,30 @@ template <typename Op>
 void visitBinaryOp(
     const BinaryOpNode<Op>* v,
     const std::string& op_str,
-    IRPrinter* printer) {
+    IRPrinter* printer,
+    bool parens = true) {
   std::ostream& os = printer->os();
-  os << "(";
+  int self_prec = getPrecedence(v->expr_type());
+  int lhs_prec = getPrecedence(v->lhs()->expr_type());
+  int rhs_prec = getPrecedence(v->rhs()->expr_type());
+
+  if (lhs_prec >= self_prec) {
```

Comment:
Don't you also need to know the direction? If it is left-to-right, and "lhs_pre == self_prec", you don't need to the "()", right?

Reply:
I would prefer not to rely on associativity directions.  I can never remember the rules for them (and they sometimes differ by language), so I think having the extra parens is probably OK there.